packages/runtime/container-runtime/src/summarizerTypes.ts

```diff
@@ -238,13 +238,10 @@ export interface ISummarizeResults {
     readonly receivedSummaryAckOrNack: Promise<SummarizeResultPart<IAckSummaryResult, INackSummaryResult>>;
 }
 
-export type OnDemandSummarizeResult = (ISummarizeResults & {
-    /** Indicates that an already running summarize attempt does not exist. */
-    readonly alreadyRunning?: undefined;
-}) | {
-    /** Resolves when an already running summarize attempt completes. */
-    readonly alreadyRunning: Promise<void>;
-};
+export interface IOnDemandSummarizeResults extends ISummarizeResults {
```

Comment:
Any chance you can add some comment on usage? Just looking at interface (and not looking at implementation details), it's hard for me to reason over composition here. I.e., if summarizerStarted resolves to failure, then I assume all other 3 promises will also fail with same reason / failure? And if it succeeded, then summary proceeded a bit further (maybe failed, maybe not). Is that right?

Seems reasonable, I wonder if callers care about this detail (i.e., API is structured or not in a way where users need to care about this detail). Some explanation here on usage patterns and pitfalls would be great!

Reply:
Yeah, I think it's probably not useful to return the promise that's related to starting the RunningSummarizer. I was following the example of ISummarizeResults, but the 3 existing promises in that interface are probably all the user needs.